cnd/src/http_api/routes/rfc003/handlers/action.rs

```diff
@@ -81,7 +93,17 @@ pub fn handle_action<C: Connect>(
                                     channel.send(response).map_err(problem::send_over_channel)?;
 
                                     let request = state.request();
-                                    con.spawn_bob(request, Err(decline_body));
+                                    let receiver =
```

Comment:
The tests are likely failing because you need to insert the initial state here before you call `spawn`, i.e. we never insert the declined state here.

Actually, for declined, we no longer need to call spawn here because nothing is happening after `decline`. That is one of the benefits of splitting up the work `spawn` is/was doing!

Reply:
Thanks for the suggestion, I do not think this is the case.  We insert the proposed state in `initiate_request()` then insert the accept/decline state in `send_request_and_spawn_alice()`.  You are right there used to be a state insert during spawn but this was already removed in the previous patch (that passes e2e tests).

Reply:
Fair point, will do some more investigation to see why the tests fail then :)